### ISSUE TITLE
Add argument to brm() for saving brmsfit objects to disk

### ISSUE THREAD

**Issue by mvuorre** [paul-buerkner/brms#472]
It would be very helpful if there was an argument in `brm()` to allow saving the resulting brmsfit object as an .rda file on disk.

Currently, there are options for saving the Stan code as a text file (`save_model`<sup>1</sup>), or the compiled model as a dso file. However, we often include multiple models in rmarkdown documents, and would therefore like to save the resulting object, including samples, to save time. Neither caching the code chunk or embedding the `brm()` call into a `if {} else {}` is optimal: The former is not available in interactive mode, the latter makes the code more difficult to read.

Therefore, it would be great if I could call something like `brm(..., file = "my_fit.rda")`. Then, if `my_fit.rda` doesn't exist, the model would be estimated and saved into that file. If the file exists, brms could simply load the brmsfit object from it.

Possibly related to issue #470 .

<sup>1</sup> The help page for `brm()` says: "Either NULL or a character string. In the latter case, the model code is saved in a file...", but this is slightly ambiguous. It would be helpful if it specified that "Either NULL or a character string. In the latter case, the model *'s Stan* code is saved in a *text* file "

**Comment by paul-buerkner** [comment]
That is indeed a good idea. Here is what I currently use in my rmarkdown files:

```R
run_model <- function(expr, path, reuse = TRUE) {
  path <- paste0(path, ".Rds")
  if (reuse) {
    fit <- suppressWarnings(try(readRDS(path), silent = TRUE))
  }
  if (is(fit, "try-error")) {
    fit <- eval(expr)
    saveRDS(fit, file = path)
  }
  fit
}
```

And then

```R
fit <- run_model(brm(...), path = "...")
```

I prefer RDS over RDA as the former is cleaner in the sense that it only stores one object which can be loaded into an object directly. Maybe I can add some functionality directly in brms.

**Comment by mvuorre** [comment]
That seems like a good approach, and I do something similar. rds seems like the better alternative, but perhaps this can be left to the user as well (automatically select the extension based on the user-provided filename, if it's not too complicated?)

But embedding the brm() call into a function makes the code harder to read and understand. Especially for beginners. So yeah if that could be automated into a brm() argument, that would be swell! But I understand if its too much of a special case...

Thanks

**Comment by paul-buerkner** [comment]
It's not too much of a special case. I actually need this all the time but was to lazy to come up with a built-in solution... Since I want brm to be able to load the stored object without side effects, using RDS is to only sensible option from my point of view.

**Comment by paul-buerkner** [comment]
Do you think it makes sense to add a new argument `file` or rather re-use argument `fit` which right know only allows to pass brmsfit objects?

**Comment by mvuorre** [comment]
I looked into RDS and agree it is best. 

I think you want to be able to re-sample if you pass a brmsfit to `fit`, so then there would have to be another argument to turn off sampling if you pass a file to `fit`, right? Considering that, `file` (or possibly, `save_file` or `save_brmsfit` to be consistent with the other `save_*` arguments) seems better.

**Comment by paul-buerkner** [comment]
I agree. The behavior when passing stuff to `fit` is much different so that adding a new argument may be less confusing to users. 

The problem with any `save_*` is that it suggests saving something not loading if it already exists. Considering that `file` seems to be better. What do you think?

**Comment by mvuorre** [comment]
Good point; best to go with `file`, since it indeeds loads and saves.

**Comment by paul-buerkner** [comment]
You can now try out the discussed feature in the dev version of brms. I would love to hear your feedback also on the argument documentation.

**Comment by mvuorre** [comment]
It works just as expected, thank you! I fixed a small typo in #473

**Comment by paul-buerkner** [comment]
Thanks! This new feature already helped me today and reduced the require code overhead in one of my analyses. So good that you brought this up and motivated me to implement it. :-) Closing this now.
